### ISSUE TITLE
Handle 'UNIQUE' validation without giving fatal error of duplicate entry.

### ISSUE THREAD

**Issue by Varun006** [proengsoft/laravel-jsvalidation#138]
How can i manage the 'unique:tablename' validation with this plugin just like any other errors.

For eg, 'required' error does not throw a fatal error like 'unique' does. I want to handle 'unique' errors just like any other errors.

Thank you in advance.

**Comment by Varun006** [comment]
Any help pls ??

**Comment by Skintillion** [comment]
'email' => 'unique:users,email_address'

    better put as

    'email' => 'unique:TABLE,FIELD'

You need a field as well.

**Comment by Varun006** [comment]
Thank Skintillion,

This did not help. I keeps on giving error 'Whoops something went wrong'.

Here is my code.

```
public function rules()
    {

            return [
                'name' => 'required',
                'email' => 'required|email|unique:users,email',
                'phone' => 'required|digits:10',
                'building_flat' => 'required',
                'street' => 'required',
                'state' => 'required',
                'city' => 'required',
                'zip' => 'required|digits:6',
                'password' => 'required|min:6',
                'confirm_password' => 'required|same:password',
                'shipping-form-name' => 'required_without:same,yes',
                'shipping-form-email' => 'required_without:same,yes',
                'shipping-form-phone' => 'required_without:same,yes',
                'shipping-building_flat' => 'required_without:same,yes',
                'shipping-street' => 'required_without:same,yes',
                'shipping-state' => 'required_without:same,yes',
                'shipping-city' => 'required_without:same,yes',
                'shipping-zip' => 'required_without:same,yes',

            ];

    }

    public function messages()
    {
            return [
              'shipping-form-name.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-form-email.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-form-phone.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-building_flat.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-street.required_without' => 'This field is required
when your billing and shipping address is not same',
              'shipping-state.required_without' => 'This field is required
when your billing and shipping address is not same',
              'shipping-city.required_without' => 'This field is required
when your billing and shipping address is not same',
              'shipping-zip.required_without' => 'This field is required
when your billing and shipping address is not same',
            ];
    }
```





*-- Varun DavdaPHP Developer*

**Comment by Varun006** [comment]
Thanks @Skintillion ,

Thank Skintillion,

This did not help. I keeps on giving error 'Whoops something went wrong'.

Here is my code.


```
public function rules()
    {

            return [
                'name' => 'required',
                'email' => 'required|email|unique:users,email',
                'phone' => 'required|digits:10',
                'building_flat' => 'required',
                'street' => 'required',
                'state' => 'required',
                'city' => 'required',
                'zip' => 'required|digits:6',
                'password' => 'required|min:6',
                'confirm_password' => 'required|same:password',
                'shipping-form-name' => 'required_without:same,yes',
                'shipping-form-email' => 'required_without:same,yes',
                'shipping-form-phone' => 'required_without:same,yes',
                'shipping-building_flat' => 'required_without:same,yes',
                'shipping-street' => 'required_without:same,yes',
                'shipping-state' => 'required_without:same,yes',
                'shipping-city' => 'required_without:same,yes',
                'shipping-zip' => 'required_without:same,yes',

            ];

    }

    public function messages()
    {
            return [
              'shipping-form-name.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-form-email.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-form-phone.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-building_flat.required_without' => 'This field is
required when your billing and shipping address is not same',
              'shipping-street.required_without' => 'This field is required
when your billing and shipping address is not same',
              'shipping-state.required_without' => 'This field is required
when your billing and shipping address is not same',
              'shipping-city.required_without' => 'This field is required
when your billing and shipping address is not same',
              'shipping-zip.required_without' => 'This field is required
when your billing and shipping address is not same',
            ];
    }

```

**Comment by Skintillion** [comment]
Error please? I need to see the  error message  to help. Make sure debug is
on if all you see is woops and no details.

**Comment by Varun006** [comment]
Thank you @Skintillion ,

Here is the  [link](www.varundavda.com/account).

Please enter email as 'varundavda006@gmail.com' and you will be able to view the error.

Just as below.

![unique](https://cloud.githubusercontent.com/assets/9958681/13025855/a156aed0-d238-11e5-807c-4898418e62b9.png)


I also went through [this ](https://github.com/proengsoft/laravel-jsvalidation/wiki/Laravel-Auth-Controller) and tried to implement without any success.

**Comment by Skintillion** [comment]
Not getting any error message. Just accepts it?

**Comment by Varun006** [comment]
I have done server side validation in my controller. Now it will redirect
to the page with old Inputs.

Still please let me know if you have better solution ..

On Sunday 14 February 2016, Skintillion <notifications@github.com> wrote:

> Not getting any error message. Just accepts it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/proengsoft/laravel-jsvalidation/issues/138#issuecomment-183824610>
> .
>


-- 




*-- Varun DavdaPHP Developer*

**Comment by Skintillion** [comment]
I used form::model personally... to know a better solution I would need to
see your code.

On Sat, Feb 13, 2016 at 9:49 PM, Varun006 <notifications@github.com> wrote:

> I have done server side validation in my controller. Now it will redirect
> to the page with old Inputs.
>
> Still please let me know if you have better solution ..
>
> On Sunday 14 February 2016, Skintillion <notifications@github.com> wrote:
>
> > Not getting any error message. Just accepts it?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/proengsoft/laravel-jsvalidation/issues/138#issuecomment-183824610
> >
> > .
> >
>
>
> --
>
>
>
>
> *-- Varun DavdaPHP Developer*
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/proengsoft/laravel-jsvalidation/issues/138#issuecomment-183825720>
> .
>



-- 
Jeremy Jackson

**Comment by torrentalle** [comment]
Can you post yor laravel log?

**Comment by fadion** [comment]
I'm getting the exact same problem with a unique email field. It just throws "Whoops, looks like something went wrong". I'm on the latest master (updated yesterday) and this is the error on log:

```
Undefined index: _jsvalidation_validate_all in [path omitted]/vendor/proengsoft/laravel-jsvalidation/src/Remote/Resolver.php:101
```

**Comment by rsantosmello** [comment]
I'm getting the same issue here :\

The response error
`{errors: "Sorry, something went wrong.", exception: "ErrorException",…}
errors
:
"Sorry, something went wrong."
exception
:
"ErrorException"
message
:
"Undefined index: _jsvalidation_validate_all"
trace
:
[{,…}, {function: "Proengsoft\JsValidation\Remote\{closure}",…}, {,…}, {,…}, {,…}, {,…}, {,…}, {,…},…]
0
:
{,…}
1
:
{function: "Proengsoft\JsValidation\Remote\{closure}",…}
args
:
["_jsvalidation", "email", [], {}]
0
:
"_jsvalidation"
1
:
"email"
2
:
[]
3
:
{}
class
:
"Proengsoft\JsValidation\Remote\Resolver"
function
:
"Proengsoft\JsValidation\Remote\{closure}"
type
:
"->"`

My Console
`
jquery.min.js:4 POST http://test.app/admin/users/store 400 (Bad Request)k.cors.a.crossDomain.send @ jquery.min.js:4n.extend.ajax @ jquery.min.js:4(anonymous function) @ jsvalidation.js:136$.extend.check @ jquery.validate.js:639$.extend.element @ jquery.validate.js:437$.extend.defaults.onkeyup @ jquery.validate.js:292delegate @ jquery.validate.js:380n.event.dispatch @ jquery.min.js:3r.handle @ jquery.min.js:3`


My Request
`    public function rules()
    {
        return [
            'first_name'=>'required',
            'surname'=>'required',
            'email'=>'required|email|unique:users,email',
            'password'=>'required|min:6|max:15',
            'user_type_id'=>'required',
        ];
    }`

**Comment by minkbear** [comment]
got the same error on laravel 5.3.24 and laravel-jsvalidation 1.3.1

**Comment by ptsilva** [comment]
Same error here.

**Comment by pulselife** [comment]
How to fix that?

**Comment by torrentalle** [comment]
This is fixed in release `v1.4.2`

**Comment by marcelogarbin** [comment]
@Varun006 Hi, have you been able to use unique for validation?
